### PR TITLE
Restructure input_pipeline and migrate to src/maxtext

### PR DESCRIPTION
**Pull request by aireenmei**
# Description
Restructure the input pipeline folder as follows:
under src/maxtext:
input_pipeline
- packing
- - prefill_packing.py 
- - sequence_packing.py 
- tokenizer.py 
- multihost_dataloading.py
- distillation_data_processing.py (prev _distillation_data_processing.py)
- grain_data_processing.py (_grain_data_processing.py)
- grain_tokenizer.py (_grain_tokenizer.py)
- hf_data_processing.py (_hf_data_processing.py)
- input_pipeline_utils.py (_input_pipeline_utils.py)
- tfds_data_processing.py (_tfds_data_processing.py)
- tfds_data_processing_c4_mlperf.py (_tfds_data_processing_c4_mlperf.py)
- input_pipeline_interface.py 
- synthetic_data_processing.py 

Makes corresponding changes in imports

# Tests

CI test

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

File: src/maxtext/input_pipeline/packing/__init__.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Copyright 2025 Google LLC
```
Comment (SurbhiJainUSC) [inline src/maxtext/input_pipeline/packing/__init__.py]: nit: `2023-2026`

**Comment by aireenmei** [comment]
> Browsed through all the files and looks like a solid restructure and relocation. Are there any edge cases which could not be covered by the CI? Thanks!

As this is mainly restructure and migration, nothing new is added, the existing CI test should be sufficient to make sure no import errors.

File: src/maxtext/input_pipeline/packing/__init__.py (hunk 1 above)
Comment (aireenmei) [reply src/maxtext/input_pipeline/packing/__init__.py]: Thanks for catching! Fixed
